assets/js/src/core/components/horizontal-scroll/horizontal-scroll.tsx

```diff
@@ -0,0 +1,154 @@
+    if (scrollContainerRef.current !== null) {
+      const { scrollLeft, scrollWidth, clientWidth } = scrollContainerRef.current
+
+      setIsAtStart(scrollLeft === 0)
+      setIsAtEnd(scrollLeft + clientWidth >= scrollWidth)
+      setScrollRequired(scrollContainerRef.current.scrollWidth > scrollContainerRef.current.clientWidth)
+      setHideElement(scrollContainerRef.current.clientWidth < 50)
+    }
+  }
+
+  useEffect(() => {
+    if (scrollContainerRef.current !== null) {
+      updateScrollState()
+      scrollContainerRef.current.addEventListener('scroll', updateScrollState)
+
+      const resizeObserver = new ResizeObserver(() => {
+        updateScrollState()
+      })
+
+      resizeObserver.observe(scrollContainerRef.current)
+
+      return () => {
+        scrollContainerRef.current?.removeEventListener('scroll', updateScrollState)
+        resizeObserver.disconnect()
+      }
+    }
+  }, [])
+
+  const startScrollingLeft = (): void => {
+    if (scrollInterval === null) {
+      const interval = setInterval(() => {
+        if (scrollContainerRef.current !== null) {
+          scrollContainerRef.current.scrollBy({ left: -scrollAmount, behavior: 'smooth' })
+        }
+      }, scrollSpeed)
+      setScrollInterval(interval)
+    }
+  }
+  const startScrollingRight = (): void => {
+    if (scrollInterval === null) {
+      const interval = setInterval(() => {
+        if (scrollContainerRef.current !== null) {
+          scrollContainerRef.current.scrollBy({ left: scrollAmount, behavior: 'smooth' })
+        }
+      }, scrollSpeed)
+      setScrollInterval(interval)
+    }
+  }
+
+  const stopScrolling = (): void => {
+    if (scrollInterval !== null) {
+      clearInterval(scrollInterval)
+      setScrollInterval(null)
+    }
+  }
+
+  const handleKeyUp = (e: React.KeyboardEvent): void => {
+    if (e.key === 'Enter' || e.key === ' ') {
+      stopScrolling()
+    }
+  }
+
+  const handleKeyDown = (e: React.KeyboardEvent, direction: 'left' | 'right'): void => {
+    if (e.key === 'Enter' || e.key === ' ') {
+      if (direction === 'left') {
+        startScrollingLeft()
+      } else if (direction === 'right') {
+        startScrollingRight()
+      }
+    }
+  }
+
+  return (
+    <Flex
+      align={ 'center' }
+      className={ styles.scrollContainer }
+      justify={ 'center' }
+    >
+      {scrollRequired && (
+        <IconButton
+          disabled={ isAtStart }
+          icon={ 'chevron-left' }
+          iconOptions={ { height: 18, width: 18 } }
+          onKeyDown={ (e) => {
+            handleKeyDown(e, 'left')
+          } }
+          onKeyUp={ handleKeyUp }
+          onMouseDown={ startScrollingLeft }
+          onMouseLeave={ stopScrolling }
+          onMouseUp={ stopScrolling }
+          theme="secondary"
+        />
+      )}
+      <Flex
+        align={ 'center' }
+        className={ styles.scroll }
+        ref={ scrollContainerRef }
+      >
+        {hideElement ? children : children}
```

Comment:
When it's either children or children. Couldn't we use children here without the condition?

Reply:
of course - this was left over I missed.
I just updated it.